### PR TITLE
📖  Add details about lifecycle/active in the documentation

### PR DESCRIPTION
**Pull request by shreyasbapat**
**What this PR does / why we need it**:

As discussed in https://github.com/kubernetes-sigs/cluster-api/issues/3315#issuecomment-658668254, the documentation does not mention about the label, hence creating some confusion on what issues to pick. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
None

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ should not need deep knowledge of the system. [Have a look and see if anything s
 interesting](https://github.com/kubernetes-sigs/cluster-api/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
 Alternatively, read some of the docs on other controllers and try to write your own, file and fix any/all issues that
 come up, including gaps in documentation!
+Also, before starting to work on the issue, make sure that it is not already taken up by someone else. The issues already taken up by someone have a [lifecycle/active](https://github.com/kubernetes-sigs/cluster-api/labels/lifecycle%2Factive) label.
```
Comment (fabriziopandini) [inline CONTRIBUTING.md:41]: What about moving this sentence before Alternatively? I think that it is more appropriate there...

Comment (fabriziopandini) [reply CONTRIBUTING.md:41]: Also, what about adding the following sentence before line 47 so we are describing this practice also when someone starts to contribute to a patch
 
1. If working on an issue,  signal to the other contributes that you starting to work on it by applying the

Comment (shreyasbapat) [reply CONTRIBUTING.md:41]: Sure.

Comment (shreyasbapat) [reply CONTRIBUTING.md:41]: Can anyone apply this label by comenting `\lifecycle active` ?

Comment (shreyasbapat) [reply CONTRIBUTING.md]: I made some changes :D

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -36,12 +36,15 @@ and instructions for signing it [can be found here](https://github.com/kubernete
 If you're new to the project and want to help, but don't know where to start, we have a semi-curated list of issues that
 should not need deep knowledge of the system. [Have a look and see if anything sounds
 interesting](https://github.com/kubernetes-sigs/cluster-api/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+Before starting to work on the issue, make sure that it is not already taken up by someone else. The issues already taken up by someone have a [lifecycle/active](https://github.com/kubernetes-sigs/cluster-api/labels/lifecycle%2Factive) label.
```
Comment (vincepri) [inline CONTRIBUTING.md:39]:
```suggestion
Before starting to work on the issue, make sure that it doesn't have a [lifecycle/active](https://github.com/kubernetes-sigs/cluster-api/labels/lifecycle%2Factive) label. If the issue has been assigned, reach out to the assignee. 
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -36,12 +36,15 @@ and instructions for signing it [can be found here](https://github.com/kubernete
...
 interesting](https://github.com/kubernetes-sigs/cluster-api/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+Before starting to work on the issue, make sure that it is not already taken up by someone else. The issues already taken up by someone have a [lifecycle/active](https://github.com/kubernetes-sigs/cluster-api/labels/lifecycle%2Factive) label. 
 Alternatively, read some of the docs on other controllers and try to write your own, file and fix any/all issues that
 come up, including gaps in documentation!
 
+
 ## Contributing a Patch
 
 1. If you haven't already done so, sign a Contributor License Agreement (see details above).
+1. If working on an issue, signal the other contributors that you are starting to work on it by leaving a comment.
```
Comment (vincepri) [inline CONTRIBUTING.md:47]:
```suggestion
1. If working on an issue, signal other contributors that you are actively to work on it using `/lifecycle active`. 
```

Comment (shreyasbapat) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 2 above)
Comment (shreyasbapat) [reply CONTRIBUTING.md]: Done

**Comment by fabriziopandini** [comment]
thanks @shreyasbapat !
/lgtm

**Pull request merged**